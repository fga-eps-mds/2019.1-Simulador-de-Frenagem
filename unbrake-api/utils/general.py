'''
For keeping general util functions to the whole project
'''

import os


def get_secret(secret_name):
    '''
    For getting project secrets which are mount on docker-compose files
    '''

    key_file = None
    try:
        key_file = open(os.path.join('/', 'run', 'secrets', secret_name))
    except FileNotFoundError:
        raise FileNotFoundError(
            f"""
            ===================================================================
            '{secret_name}' secret not found.
            Ask the development team the secrets or use your owns.
            ===================================================================
            """
        )

    return key_file.read().strip()
