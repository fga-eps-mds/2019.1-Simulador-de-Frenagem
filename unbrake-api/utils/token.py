'''
    This fixture create a user and a token to be used in te test
'''
import pytest
from graphql_jwt.shortcuts import get_token
from django.contrib.auth import get_user_model


@pytest.fixture
@pytest.mark.django_db
def create_token(is_superuser=False):
    '''
        Save a user on db and return the token
    '''
    def _create_token(is_superuser=is_superuser):

        user = get_user_model()(
            username="Teste",
            email="mail@mail.com",
            is_superuser=is_superuser
        )
        user.set_password("abobora")
        user.save()

        token = get_token(user)
        return "JWT " + token
    return _create_token
