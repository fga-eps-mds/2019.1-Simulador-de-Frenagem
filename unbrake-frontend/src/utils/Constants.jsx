export const API_URL = process.env.REACT_APP_API_URL;
export const API_URL_GRAPHQL = process.env.REACT_APP_API_URL;
export const MQTT_HOST = process.env.REACT_APP_MQTT_HOST;
export const MQTT_PORT = process.env.REACT_APP_MQTT_PORT;
export const base10 = 10;

export const conversionFunction = (x, a, b) => {
  return parseFloat(a) * parseFloat(x) + parseFloat(b);
};
