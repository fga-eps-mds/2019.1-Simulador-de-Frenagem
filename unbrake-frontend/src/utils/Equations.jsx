const validNumber = 0;
const decimalPlace = 4;
const maximumVoltage = 5000;
const megaByte = 1023;

const frequencySubtractionFactor = 826;
const frequencyDivisionFactor = 15;
const minutes = 60;

const tireDivisionFactor = 100000;
const tireDivision = 2;
const tireMultiplication = 0.0254;

const completeTurn = tireDivision * Math.PI;
const conversionSpeedFactor = 3.6;

export const base10 = 10;

export const linearEquation = (x, a, b) => {
  return parseFloat(a) * parseFloat(x) + parseFloat(b);
};

export const convertDigitalToAnalog = digitalSignal => {
  const analogSignal = (parseFloat(digitalSignal) * maximumVoltage) / megaByte;
  return analogSignal.toFixed(decimalPlace);
};

export const frequencyEquation = analogSignal => {
  const frequency =
    (parseFloat(analogSignal) - frequencySubtractionFactor) /
    frequencyDivisionFactor;
  return frequency;
};

export const rotationsPerMinuteEquation = frequency => {
  const rotationsPerMinute = parseFloat(frequency * minutes);
  return rotationsPerMinute;
};

export const tireRadiusEquation = (
  crossSectionalWidth,
  widthHeightRatio,
  arcDiameter
) => {
  const LST = parseFloat(crossSectionalWidth);
  const RAL = parseFloat(widthHeightRatio);
  const DIA = parseFloat(arcDiameter);

  if (LST > validNumber && RAL > validNumber && DIA > validNumber) {
    const tireRadius =
      (LST * RAL) / tireDivisionFactor +
      (tireMultiplication * DIA) / tireDivision;
    return tireRadius.toFixed(decimalPlace);
  }
  return undefined;
};

export const gearRatioEquation = (
  synchronousMotorRotation,
  diameterPowerPulley,
  diameterMovedPulley
) => {
  const RSM = parseFloat(synchronousMotorRotation);
  const DPO = parseFloat(diameterPowerPulley);
  const DPM = parseFloat(diameterMovedPulley);

  if (RSM > validNumber && DPO > validNumber && DPM > validNumber) {
    const gearRatio = (RSM * DPO) / DPM;
    return gearRatio.toFixed(decimalPlace);
  }
  return undefined;
};

export const topSpeedEquation = (tireRadius, gearRatio) => {
  const RDP = parseFloat(tireRadius);
  const RDT = parseFloat(gearRatio);

  const topSpeed = (RDP * conversionSpeedFactor * completeTurn * RDT) / minutes;
  return topSpeed.toFixed(decimalPlace);
};
