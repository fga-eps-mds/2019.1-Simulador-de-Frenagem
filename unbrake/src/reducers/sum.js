const sum = (state = {}, action) => { //Jeito de definir uma função
  switch (action.type) {
    case 'ADD':
      console.log(state, action)
      return {...state, sum: +action.n1 + +action.n2};
      
    default:
      return state
  }
}

export default sum 
