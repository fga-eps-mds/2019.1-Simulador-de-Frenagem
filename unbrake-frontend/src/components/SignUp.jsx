import React from "react";
import { Field, reduxForm, SubmissionError } from "redux-form";
import { TextField } from "redux-form-material-ui";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import { withStyles } from "@material-ui/core/styles";
import PropTypes from "prop-types";
import history from "../utils/history";
import { API_URL_GRAPHQL } from "../utils/Constants";

const padding = 10;

const styles = theme => ({
  root: {
    flexGrow: 1
  },
  paper: {
    padding: theme.spacing.unit * padding,
    textAlign: "center",
    color: theme.palette.text.secondary,
    width: "600px"
  },
  field: {
    paddingTop: theme.spacing.unit * padding,
    paddingBottom: theme.spacing.unit * padding
  },
  grid: {
    padding: "5px"
  }
});

export const validate = values => {
  const errors = {};
  const requiredFields = ["username", "password", "confirmPassword"];
  requiredFields.forEach(field => {
    if (!values[field]) {
      errors[field] = "Obrigatório";
    } else if (values.password !== values.confirmPassword) {
      errors.confirmPassword = "Não confere";
    }
  });
  return errors;
};

const renderField = ({ input, label, type, meta: { touched, error } }) => (
  <div>
    <TextField
      label={label}
      placehoder={label}
      error={error && touched}
      type={type}
      variant="outlined"
      fullWidth
      helperText={touched && error}
      {...input}
    />
  </div>
);

const signUpButton = (classes, submitting) => {
  return (
    <Grid container item xs={12} alignItems="center" justify="center">
      <Grid item xs={3} className={classes.grid}>
        <Button
          color="secondary"
          fullWidth
          variant="contained"
          type="submit"
          disabled={submitting}
        >
          Registrar
        </Button>
      </Grid>
    </Grid>
  );
};

export const submit = values => {
  return fetch(
    `${API_URL_GRAPHQL}?query=mutation{createUser(password: "${
      values.password
    }",
     username: "${values.username}"){user{id}}}`,
    {
      method: "POST"
    }
  )
    .then(response => {
      return response.json();
    })
    .then(parsedData => {
      if (parsedData.errors !== undefined) {
        if (
          parsedData.errors[0].message ===
          "UNIQUE constraint failed: auth_user.username"
        ) {
          throw new SubmissionError({
            username: "O usuário já está em uso",
            _error: "Login failed!"
          });
        }
      } else if (parsedData.data.createUser.user !== null) {
        history.push("/login");
      }
    });
};
const signUpPaper = (classes, handleSubmit, submitting) => {
  return (
    <Paper className={classes.paper} elevation={10}>
      <Typography variant="h4" color="secondary" className={classes.grid}>
        Registre-se
      </Typography>
      <form onSubmit={handleSubmit(submit)}>
        <Grid className={classes.grid}>
          <Field
            name="username"
            type="text"
            component={renderField}
            label="Usuário"
            variant="outlined"
            fullWidth
            className={classes.field}
          />
        </Grid>
        <Grid item xs={12} sm={12} className={classes.grid}>
          <Field
            name="password"
            type="password"
            component={renderField}
            label="Senha"
            variant="outlined"
            className={classes.field}
          />
        </Grid>
        <Grid item xs={12} sm={12} className={classes.grid}>
          <Field
            name="confirmPassword"
            type="password"
            component={renderField}
            label="Confirme a Senha"
            variant="outlined"
            className={classes.field}
          />
        </Grid>
        {signUpButton(classes, submitting)}
      </form>
    </Paper>
  );
};

class SignUp extends React.PureComponent {
  render() {
    const { classes, handleSubmit, submitting } = this.props;
    return (
      <Grid
        alignItems="center"
        justify="center"
        style={{ minHeight: "100vh" }}
        container
        spacing={16}
      >
        {signUpPaper(classes, handleSubmit, submitting)}
      </Grid>
    );
  }
}

renderField.propTypes = {
  input: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.func, PropTypes.string])
  ).isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.func, PropTypes.bool, PropTypes.string])
  ).isRequired
};

SignUp.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired
};
const SignUpForm = reduxForm({
  form: "signup",
  validate
})(SignUp);

export default withStyles(styles)(SignUpForm);
