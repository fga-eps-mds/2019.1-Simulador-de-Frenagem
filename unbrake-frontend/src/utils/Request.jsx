import Cookies from "universal-cookie";
import { API_URL_GRAPHQL } from "./Constants";

const cookies = new Cookies();

const invalidId = -1;

export default async function Request(url, method) {
  const token = cookies.get("token");
  const responser = await fetch(url, {
    headers: { Authorization: `JWT ${token}` },
    method
  }).then(response => {
    return response.json();
  });

  return responser;
}

export const createMutationUrl = async (
  informations,
  variableNames,
  values
) => {
  const initialUrl = `${API_URL_GRAPHQL}?query=mutation{${
    informations.mutation
  }(`;

  let url = variableNames.reduce((prevMessage, newField) => {
    if (prevMessage.length === initialUrl.length) {
      if (Number.isNaN(Number(values[newField.front]))) {
        return `${prevMessage}${newField.back}:"${values[newField.front]}"`;
      }
      return `${prevMessage}${newField.back}:${values[newField.front]}`;
    }
    if (Number.isNaN(Number(values[newField.front]))) {
      return `${prevMessage},${newField.back}:"${values[newField.front]}"`;
    }
    return `${prevMessage},${newField.back}:${values[newField.front]}`;
  }, initialUrl);

  url += `){${informations.response}{${informations.variablesResponse}}}}`;

  const response = await Request(url, "POST");

  if (response.errors !== undefined)
    return { invalidId, error: response.error };

  return response.data[informations.mutation][informations.response][
    informations.variablesResponse
  ];
};
