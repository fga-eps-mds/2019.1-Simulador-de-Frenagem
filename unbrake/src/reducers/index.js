import { combineReducers } from 'redux'
import sum from './sum'

const Reducer = combineReducers({ //Agrupa os componentes
	sum,
})

export default Reducer;