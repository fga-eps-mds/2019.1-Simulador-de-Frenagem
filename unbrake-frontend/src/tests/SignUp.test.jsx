import React from "react";
import Enzyme, { shallow } from "enzyme";
import { Provider } from "react-redux";
import { createStore } from "redux";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import fetch from "jest-fetch-mock";
import SignUp, {
  validate,
  submit,
  required,
  validateEmail
} from "../components/SignUp";
import { API_URL_GRAPHQL } from "../utils/Constants";
import reducers from "../reducer/index";

Enzyme.configure({ adapter: new Adapter() });

describe("<SignUp />", () => {
  describe("render()", () => {
    test("renders the component", () => {
      const context = React.createContext();
      const store = createStore(reducers);
      const wrapper = shallow(
        <Provider context={context} store={store}>
          <SignUp />
        </Provider>
      );
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });

  test("validate()", () => {
    const validate1 = {
      password: "password",
      confirmPassword: "apassword"
    };
    const validate2 = {
      password: "password",
      confirmPassword: "password"
    };
    const error1 = validate(validate1);
    const error2 = validate(validate2);

    expect(error1).toEqual({ confirmPassword: "As senhas não são iguais" });
    expect(error2).toEqual({});
  });

  test("required()", () => {
    const required1 = undefined;
    const required2 = "teste";

    const error1 = required(required1);
    const error2 = required(required2);
    expect(error1).toEqual("Obrigatório");
    expect(error2).toEqual();
  });

  test("validateEmail()", () => {
    const validateEmail1 = "teste";
    const validateEmail2 = "teste@teste.com";

    const error1 = validateEmail(validateEmail1);
    const error2 = validateEmail(validateEmail2);
    expect(error1).toEqual("Endereço de email inválido");
    expect(error2).toEqual();
  });
});

describe("submit", () => {
  it("Call request and return response", () => {
    beforeEach(() => {
      fetch.resetMocks();
    });
    const values = { username: "username", password: "password" };
    let response = "";
    fetch.mockResponseOnce(
      JSON.stringify({ data: { createUser: { username: values.username } } })
    );

    submit(values)
      .then(res => {
        response = res.data.createUser.username;
        expect(response).toBe(values.username);
        expect(fetch.mock.calls[0][0]).toEqual(API_URL_GRAPHQL);
      })
      .catch(() => {});
  });
});
