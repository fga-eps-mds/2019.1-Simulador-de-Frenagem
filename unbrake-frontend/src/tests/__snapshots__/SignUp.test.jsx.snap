// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> render() renders the component 1`] = `
<Hoc
  classes={
    Object {
      "field": "ReduxForm-field-3",
      "grid": "ReduxForm-grid-4",
      "paper": "ReduxForm-paper-2",
      "root": "ReduxForm-root-1",
    }
  }
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="signup"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
